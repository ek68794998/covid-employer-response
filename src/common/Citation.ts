import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
