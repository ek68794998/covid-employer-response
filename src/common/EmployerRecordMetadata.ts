import { EmployerRating } from "./EmployerRating";
import { EmployerRecordBase } from "./EmployerRecordBase";

export class EmployerRecordMetadata extends EmployerRecordBase {
	public readonly negativeCount: number;

	public readonly positiveCount: number;

	public readonly ratingValue: EmployerRating;

	public constructor(
		negativeCount: number,
		positiveCount: number,
		ratingValue: EmployerRating) {

		super();

		this.negativeCount = negativeCount;
		this.positiveCount = positiveCount;
		this.ratingValue = ratingValue;
	}

	public get rating(): EmployerRating {
		return this.ratingValue;
	}

	public static getTrendIcon(e: EmployerRecordMetadata): "trending_up" | "trending_flat" | "trending_down" {
		switch (e.rating) {
			case "good":
				return "trending_up";

			case "poor":
				return "trending_down";

			case "fair":
			default:
				return "trending_flat";
		}
	}
}
