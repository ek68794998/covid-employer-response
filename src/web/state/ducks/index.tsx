import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});