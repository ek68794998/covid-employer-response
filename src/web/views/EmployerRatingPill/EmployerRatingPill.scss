@import "../Colors.scss";
@import "../Dimensions.scss";

$padding-interior: 12px;

.EmployerRatingPill__Box {
	background: $indicator-neutral;
	border-radius: 16px;
	color: #fff;
	cursor: default;
	font: 600 18px 'Baloo Chettan 2';
	height: 32px;
	margin-left: $padding-interior;
	padding: 3px 12px 13px 12px;

	&.EmployerRatingPill__Box--good {
		background: $indicator-positive;
	}

	&.EmployerRatingPill__Box--poor {
		background: $indicator-negative;
	}

	&.EmployerRatingPill__Box--LowContrast {
		font-size: 24px;
		font-weight: 400;
		padding: 0;

		&.EmployerRatingPill__Box--fair {
			background: none;
			color: $indicator-neutral;
		}
	
		&.EmployerRatingPill__Box--good {
			background: none;
			color: $indicator-positive;
		}
	
		&.EmployerRatingPill__Box--poor {
			background: none;
			color: $indicator-negative;
		}
	}

	.material-icons {
		margin-left: $padding-interior;
		user-select: none;
		vertical-align: bottom;
	}
}
