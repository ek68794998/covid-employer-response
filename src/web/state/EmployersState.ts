import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}