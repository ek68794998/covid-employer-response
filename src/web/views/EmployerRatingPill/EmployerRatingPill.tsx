import React from "react";
import { useSelector } from "react-redux";
import { RouteProps } from "react-router-dom";

import { DesignHelpers } from "../../../common/DesignHelpers";
import { EmployerRating } from "../../../common/EmployerRating";
import { EmployerRecordMetadata } from "../../../common/EmployerRecordMetadata";
import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { getStrings } from "../../state/ducks/localization/selectors";

import "./EmployerRatingPill.scss";

interface Props extends RouteProps {
	employer: EmployerRecordMetadata;

	isAnnotated?: boolean;

	isLowContrast?: boolean;

	showGrade?: boolean;
}

const EmployerRatingPill: React.FC<Props> = (props: Props): React.ReactElement | null => {
	const strings: LocalizedStrings = useSelector(getStrings);
	const { employer, isAnnotated, isLowContrast, showGrade } = props;

	let text: string;
	let classSubtype: EmployerRating;

	if (showGrade) {
		classSubtype = employer.score > 0 ? "good" : employer.score < 0 ? "poor" : "fair";
		text = `${employer.score > 0 ? "+" : ""}${Math.round(employer.score)}`;
	} else {
		classSubtype = employer.rating;
		text = strings.ratingLabels[employer.rating];
	}

	return (
		<span
			className={`EmployerRatingPill__Box EmployerRatingPill__Box--${classSubtype} ${isLowContrast ? "EmployerRatingPill__Box--LowContrast" : ""}`}
			title={strings.detailDescriptions.rating}
		>
			{text}
			{isAnnotated && "*"}
			{!showGrade && DesignHelpers.materialIcon(EmployerRecordMetadata.getTrendIcon(employer))}
		</span>
	);
};

export default EmployerRatingPill;
