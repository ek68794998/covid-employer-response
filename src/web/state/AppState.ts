import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}