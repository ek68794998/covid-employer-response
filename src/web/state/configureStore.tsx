import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;