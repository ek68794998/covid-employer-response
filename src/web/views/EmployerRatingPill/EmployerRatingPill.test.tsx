import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { getPlocStringsAsync } from "../../../__tests__/TestUtils";
import { EmployerRecordMetadata } from "../../../common/EmployerRecordMetadata";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerRatingPill from "./EmployerRatingPill";

describe("<EmployerRatingPill />", () => {
	test("renders without exploding", async () => {
		const store: Store<AppState, AnyAction> = configureStore({ strings: await getPlocStringsAsync() });

		const employer: EmployerRecordMetadata =
			new EmployerRecordMetadata(3, 1, 0, "good");

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerRatingPill employer={employer} isAnnotated={true} />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
