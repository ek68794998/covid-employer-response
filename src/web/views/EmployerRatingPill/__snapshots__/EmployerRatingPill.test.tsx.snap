// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerRatingPill /> renders without exploding 1`] = `
<span
  className="EmployerRatingPill__Box EmployerRatingPill__Box--good "
  title="Tλè ϱèñèřáℓ řáƭïñϱ ôƒ ƭλè è₥ƥℓô¥èř."
>
  Gôôδ
  *
  <i
    className="material-icons"
  >
    trending_up
  </i>
</span>
`;
